External/FEXCore/Source/Interface/Core/JIT/Arm64/MemoryOps.cpp

```diff
@@ -1340,6 +1340,165 @@ DEF_OP(StoreMemTSO) {
   }
 }
 
+DEF_OP(MemSet) {
+  const auto Op = IROp->C<IR::IROp_MemSet>();
+
+  const int32_t Size = Op->Size;
+  const auto MemReg = GetReg(Op->Addr.ID());
+  const auto Value = GetReg(Op->Value.ID());
+  const auto Length = GetReg(Op->Length.ID());
+  const auto Direction = GetReg(Op->Direction.ID());
+  const auto Dst = GetReg(Node);
+
+  // If Direction == 0 then:
+  //   MemReg is incremented (by size)
+  // else:
+  //   MemReg is decremented (by size)
+  //
+  // Counter is decremented regardless.
+
+  ARMEmitter::ForwardLabel BackwardImpl{};
+  ARMEmitter::ForwardLabel Done{};
+
+  mov(TMP1, Length.X());
+  if (Op->Prefix.IsInvalid()) {
+    mov(TMP2, MemReg.X());
+  }
+  else {
+    const auto Prefix = GetReg(Op->Prefix.ID());
+    add(TMP2, Prefix.X(), MemReg.X());
+  }
+
+  // Backward or forwards implementation depends on flag
+  cbnz(ARMEmitter::Size::i64Bit, Direction, &BackwardImpl);
+
+  auto MemStore = [&](auto OpSize, int32_t Size) {
+    switch (OpSize) {
+      case 1:
+        strb<ARMEmitter::IndexType::POST>(Value.W(), TMP2, Size);
+        break;
+      case 2:
+        strh<ARMEmitter::IndexType::POST>(Value.W(), TMP2, Size);
+        break;
+      case 4:
+        str<ARMEmitter::IndexType::POST>(Value.W(), TMP2, Size);
+        break;
+      case 8:
+        str<ARMEmitter::IndexType::POST>(Value.X(), TMP2, Size);
+        break;
+      default:
+        LOGMAN_MSG_A_FMT("Unhandled {} size: {}", __func__, Size);
+        break;
+    }
+  };
+
+  auto MemStoreTSO = [&](auto OpSize, int32_t Size) {
```

Comment:
If possible, a `this` capture restricts what state can leak into this lambda:
```suggestion
  auto MemStoreTSO = [this](auto OpSize, int32_t Size) {
```

Reply:
restricted.